### PR TITLE
Added marking add-ons as deprecated

### PR DESCRIPTION
**Pull request by kyleprice**

### CODE REVIEW

**Review by AmberAlston — CHANGES_REQUESTED** [review]
Lots of comments. Let me know if you want to discuss any of them in more depth outside of async comments.

Visuals/Content:
- The Deprecated label is really hard to read unless you’re on a high-res monitor. Maybe it’s the length of the word or the color. We could try using the existing beta red. 
- Why are we stating the start date for deprecation (March 23) on some of the add-ons but not others? 
- Why is Antrea June 1, and AWS is May 1. Latter is beta, but both don’t appear to have any usage
- How did we reach the date of Sept 1 for the others? That’s approx 5mo. Is that enough time for an add-on like Weave, which is highly used today?
- For Weave specifically we should say something in the deprecation notice about the upstream maintainers no longer maintaining the project, and as such we can no longer maintain it as a viable CNI add-on for KURL. 
- For Docker specifically we should highlight that “Kubernetes no longer supports Docker as of 1.24, and therefore any in-support versions of Kubernetes must leverage ContainerD”. That way is reinforces that this isn’t our decision. We say something similar later, but it should be in the deprecation notice. 
- We need to change this language in the messages, “Support for existing installs will continue to be best effort.”, it sounds like we’re abandoning our support responsibilities. I’d suggest something like, “Existing installs that use this add-on will continue to be best-effort supported until the deprecation date.”  Are we continuing to patch these add-ons during this window? That answer may drive some additional copy tweaks 

Other related changes needed:
- If we are making Longhorn as deprecated we should clean up these example specs that say to use Longhorn: https://kurl.sh/docs/install-with-kurl/#latest & https://kurl.sh/docs/create-installer/
- The IPv6 alpha https://kurl.sh/docs/install-with-kurl/ipv6 only works with Antrea. How are we reconciling that as we mark Antrea deprecated? 

Non-code complimentary work to address before this is merged:
- What are we doing to proactively communicate this deprecation to vendors using Longhorn, Docker, and Weave. I don’t want them to stumble across the deprecation notices on the website and ask us about it. These needs to be paired with a formalized announcement to vendors impacted by these changes to let them know: (1) about the deprecation, (2) how they can move forward 
- We also need to let the field facings teams (Sales, SE, CRE) know ahead of pushing this change live, so they are prepared for possible questions.

File: src/markdown-pages/add-ons/longhorn.md
Comment (camilamacedo86) [inline src/markdown-pages/add-ons/longhorn.md:15]: What fully  vs partial deprecated does means? 
It seems that you want say it is deprecated now and at September 31st goes in EOL.

**Review by AmberAlston — CHANGES_REQUESTED** [review]
Updated comments below
* Re-reading this again today, I’d suggest we tweak the main deprecation copy as follows. This would help address Camila’s feedback as well. 
    * As of March 27, 2023, kURL no longer intends to continue to offer this add-on as part of the ongoing kURL project. This add-on is now considered deprecated, and may no longer be offered by the project after (DATE). Existing installs that use this add-on will be best effort supported during this deprecation window. (MIGRATION GUIDANCE & ANY OTHER INFO)   
* I didn’t get an answer on my question about patching updates during this deprecation window or not. That answer may drive some additional copy tweaks. 
* Docker - I’d still like to see a reference to the K8s incompatibility within the deprecation notice itself. That way it’s clearer this isn’t our decision, it’s deprecated by virtue of K8s moving on from it. 
* It looks like the reference to longhorn was corrected in the "latest" link, but not https://kurl.sh/docs/install-with-kurl/#latest
* IPv6- I feel like we should just remove the page. It’s in alpha, and we’re not aware of any one using it. By marking it as deprecated we risk implying that we’ve decided we don’t intend to support IPv6 at all. In actuality, we want signals that people want this so we can consider it with Flannel. Marking as deprecated may discourage that signal. If we remove we should include in the SE notification portion of my “Non-code complimentary work to address before this is merged” part of my prior comment.

File: src/markdown-pages/add-ons/antrea.md
Comment (emosbaugh) [inline src/markdown-pages/add-ons/antrea.md:15]:
```suggestion
kURL will not be moving forward with support of this add-on starting March 23, 2023 and will be fully removed after June 31st 2023.  Existing installs that use this add-on will continue to be best-effort supported until the removal date.  Please consider using the [Flannel](https://kurl.sh/docs/add-ons/flannel) add-on for your CNI needs moving forward.
```

File: src/markdown-pages/add-ons/aws.md
Comment (emosbaugh) [inline src/markdown-pages/add-ons/aws.md:15]:
```suggestion
kURL will not be moving forward with support of this add-on starting March 23, 2023 and will be fully removed after April 31st 2023.
```

File: src/markdown-pages/add-ons/docker.md
Comment (emosbaugh) [inline src/markdown-pages/add-ons/docker.md:14]:
```suggestion
kURL will not be moving forward with support of this add-on starting March 23, 2023 and will be fully removed after September 31st 2023.  Existing installs that use this add-on will continue to be best-effort supported until the removal date.  Please remove Docker from your spec and replace it with [Containerd](https://kurl.sh/docs/add-ons/containerd) to migrate existing installs.
```

File: src/markdown-pages/add-ons/docker.md
Comment (emosbaugh) [inline src/markdown-pages/add-ons/docker.md:23]:
```suggestion
Kubenetes 1.24.0+ no loner supports Dockershim, therefore you must use an alternative CRI such as [containerd](/docs/add-ons/containerd).
```

File: src/markdown-pages/add-ons/longhorn.md
Comment (emosbaugh) [inline src/markdown-pages/add-ons/longhorn.md:15]:
```suggestion
kURL will not be moving forward with support of this add-on starting March 23, 2023 and will be fully removed after September 31st, 2023.  Existing installs that use this add-on will continue to be best-effort supported until the removal date.  Please consider using the [OpenEBS](https://kurl.sh/docs/add-ons/openebs) or [Rook](https://kurl.sh/docs/add-ons/rook) add-on for your CSI needs moving forward.  We offer a migration path for existing customer installs as described [here](https://kurl.sh/docs/install-with-kurl/migrating-csi).
```

Comment (emosbaugh) [reply src/markdown-pages/add-ons/longhorn.md:15]: Can we use the term remove rather than deprecate?

File: src/markdown-pages/add-ons/weave.md
Comment (emosbaugh) [inline src/markdown-pages/add-ons/weave.md:15]:
```suggestion
kURL will not be moving forward with support of this add-on starting March 23, 2023 and will be fully removed after September 31st 2023.  This add-on is no longer supported by the upstream maintainers. Existing installs that use this add-on will continue to be best-effort supported until the removal date.  Please consider using the [Flannel](https://kurl.sh/docs/add-ons/flannel) add-on for your CNI needs moving forward.  We offer a migration path for existing customer installs as described [here](https://kurl.sh/docs/add-ons/flannel#migration-from-weave).
```

File: src/markdown-pages/install-with-kurl/ipv6.md
Comment (emosbaugh) [inline src/markdown-pages/install-with-kurl/ipv6.md:13]:
```suggestion
kURL will not be moving forward with support of ipv6 starting March 23, 2023 and will be fully removed after June 31st 2023.  Existing installs that use this add-on will continue to be best-effort supported until the removal date.  This is due to the deprecation of the Antrea add-on.
```

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: src/markdown-pages/add-ons/longhorn.md
Comment (camilamacedo86) [inline src/markdown-pages/add-ons/longhorn.md:17]: Would be here no longer offered.

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: src/markdown-pages/add-ons/antrea.md
Comment (camilamacedo86) [inline src/markdown-pages/add-ons/antrea.md:15]:
```suggestion
Antrea add-on is deprecated, and may no longer be offered by the project after June 31st, 2023. Existing installs that use this add-on will be best effort supported during this deprecation window. Please consider using the [Flannel](https://kurl.sh/docs/add-ons/flannel) add-on for your CNI needs moving forward.
```
What about we try to make it less verbose?
Do we need the first part?

File: src/markdown-pages/add-ons/aws.md
Comment (camilamacedo86) [inline src/markdown-pages/add-ons/aws.md:15]:
```suggestion
The AWS add-on is deprecated, and may no longer be offered by the project after April 31st, 2023. Existing installs that use this add-on will be best effort supported during this deprecation window.
```

File: src/markdown-pages/add-ons/longhorn.md
Comment (camilamacedo86) [inline src/markdown-pages/add-ons/longhorn.md:15]:
```suggestion
The Longhorn is deprecated, and may no longer be offered by the project after September 31st, 2023. Existing installs that use this add-on will be best effort supported during this deprecation window. Please consider using the [OpenEBS](https://kurl.sh/docs/add-ons/openebs) or [Rook](https://kurl.sh/docs/add-ons/rook) add-on for your CSI needs moving forward. We offer a migration path for existing customer installs as described [here](https://kurl.sh/docs/install-with-kurl/migrating-csi).
```

File: src/markdown-pages/add-ons/weave.md
Comment (camilamacedo86) [inline src/markdown-pages/add-ons/weave.md:15]:
```suggestion
The Weave add-on is deprecated, and may no longer be offered by the project after September 31st, 2023. Existing installs that use this add-on will be best effort supported during this deprecation window. Please consider using the [Flannel](https://kurl.sh/docs/add-ons/flannel) add-on for your CNI needs moving forward. We offer a migration path for existing customer installs as described [here](https://kurl.sh/docs/add-ons/flannel#migration-from-weave).
```

**Review by camilamacedo86 — COMMENTED** [review]
Just some nits, otherwise it shows great @kyleprice.
Terrific work 🥇 and I am so happy to see this "deprecated" badges  :-P

File: src/markdown-pages/add-ons/antrea.md
Comment (rrpolanco) [reply src/markdown-pages/add-ons/antrea.md:15]: I prefer having the starting date for the deprecation. We can point to this and say to existing vendors that they had ~90 day notice to transition and make changes to their spec.

File: src/markdown-pages/add-ons/docker.md
Comment (rrpolanco) [reply src/markdown-pages/add-ons/docker.md:14]: Why? The March 27th date marks the beginning of our deprecation period.

**Comment by AmberAlston** [comment]
@kyleprice In addition to myself, I'd like @crdant to help review the draft announcement communication to impacted vendors letting them know about these deprecations. He'll likely have input on the "how they can move forward" part and how we want to handle any resulting customer questions resulting from the deprecation announcement. 

> These changes need to be paired with a formalized announcement to vendors impacted by these changes to let them know: (1) about the deprecation, (2) how they can move forward

**Comment by AmberAlston** [comment]
@kyleprice In addition to myself, I'd like @crdant to help review the draft announcement communication to impacted vendors letting them know about these deprecations. He'll likely have input on the "how they can move forward" part and how we want to handle any resulting customer questions resulting from the deprecation announcement. 

> These changes need to be paired with a formalized announcement to vendors impacted by these changes to let them know: (1) about the deprecation, (2) how they can move forward

**Comment by AmberAlston** [comment]
@kyleprice In addition to myself, I'd like @crdant to also review the draft announcement to vendors about these deprecations. He'll likely have input on the "how they can move forward" and how we want to handle any resulting customer questions from this deprecation announcement. 

> These needs to be paired with a formalized announcement to vendors impacted by these changes to let them know: (1) about the deprecation, (2) how they can move forward

**Review by emosbaugh — DISMISSED** [review]
this looks good once we are comfortable with the flannel add-on

**Review by AmberAlston — APPROVED** [review]

**Comment by AmberAlston** [comment]
@replicatedhq/replicated-docs - I wanted to make sure you were aware of this pr since I don't see you in the review process yet. The team is planning to push it live in the next couple of days. I have been involved in the review so far on wording, etc.

**Review by paigecalvert — DISMISSED** [review]
done with docs edits @kyleprice !

**Review by rrpolanco — APPROVED** [review]

**Review by laverya — APPROVED** [review]

**Pull request merged**
